[CI] RealmsTests testDisabledRealmsAreNotAdded

**Build scan**:
https://gradle-enterprise.elastic.co/s/hkbv57gppt62i
https://gradle-enterprise.elastic.co/s/ax54xsy3dvan4

**Repro line**:
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.authc.RealmsTests.testDisabledRealmsAreNotAdded" \
  -Dtests.seed=26D54A457F1571D8 \
  -Dtests.security.manager=true \
  -Dtests.locale=th \
  -Dtests.timezone=America/Halifax \
  -Druntime.java=8
```

**Reproduces locally?**:
Yes

**Applicable branches**:
7.x 

**Failure history**:
2 today

**Failure excerpt**:
```
java.lang.AssertionError: unexpected warning headers
Expected: an empty collection
     but: <[299 Elasticsearch-7.13.0-SNAPSHOT-f15c31f846d0f3951c4913fba1d5cfdc313b4d7d "Found explicitly disabled basic realms: [file,native]. But they will be enabled because no other realms are configured or enabled. In next major release, explicitly disabled basic realms will remain disabled."]>
	at __randomizedtesting.SeedInfo.seed([809D7B2D3C3BEA97:86A181955AC7CE7]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.test.ESTestCase.ensureNoWarnings(ESTestCase.java:427)
```

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java
@@ -493,7 +493,6 @@ public class RealmsTests extends ESTestCase {
         assertThat(realm.name(), equalTo("foo"));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/70378")
     public void testDisabledRealmsAreNotAdded() throws Exception {
         Settings.Builder builder = Settings.builder()
                 .put("path.home", createTempDir());
@@ -504,9 +503,11 @@ public class RealmsTests extends ESTestCase {
         }
         Collections.shuffle(orders, random());
         Map<Integer, Integer> orderToIndex = new HashMap<>();
+        boolean anyEnabled = false;
         for (int i = 0; i < randomRealmTypesCount; i++) {
             builder.put("xpack.security.authc.realms.type_" + i + ".realm_" + i + ".order", orders.get(i));
             boolean enabled = randomBoolean();
+            anyEnabled = anyEnabled || enabled;
             builder.put("xpack.security.authc.realms.type_" + i + ".realm_" + i + ".enabled", enabled);
             if (enabled) {
                 orderToIndex.put(orders.get(i), i);
@@ -516,6 +517,11 @@ public class RealmsTests extends ESTestCase {
         Settings settings = builder.build();
         Environment env = TestEnvironment.newEnvironment(settings);
         Realms realms = new Realms(settings, env, factories, licenseState, threadContext, reservedRealm);
+        if (false == anyEnabled) {
+            assertWarnings("Found explicitly disabled basic realms: [file,native]. " +
+                "But they will be enabled because no other realms are configured or enabled. " +
+                "In next major release, explicitly disabled basic realms will remain disabled.");
+        }
         Iterator<Realm> iterator = realms.iterator();
         Realm realm = iterator.next();
         assertThat(realm, is(reservedRealm));

